src/Compilers/Core/CodeAnalysisTest/Collections/TemporaryArrayTests.cs

```diff
@@ -205,6 +206,69 @@ public void TestReverseContents([CombinatorialRange(0, 6)] int initialItems)
                 Assert.Equal(array[i], initialItems - 1 - i);
         }
 
+        [Fact]
+        public void TestSort()
```

Comment:
i would take everything out related to sorting TempArray.  We don't even relaly need temparray for block-structure.  That's an array that's good when you commonly have <4 elements.  But for block structure, we'd almost alwayshave more.  So we can just pass an ArrayBuilder along instead).

Reply:
or, we can keep this.  i'm on the fence.

Reply:
@CyrusNajmabadi I had the same feeling the TemporaryArray might not be very suitable for this scenario, but I preferred to limit the scope of my changes.